// ng2-sharebuttons/share-buttons.module.js
import { NgModule } from '@angular/core';
import { JsonpModule } from '@angular/http';
import { CommonModule } from '@angular/common';
import { ShareButtonsComponent } from './components/share-buttons/share-buttons.component';
import { ShareButtonComponent } from './components/share-button/share-button.component';
import { ShareButtonDirective } from './directives/share-button/share-button.directive';
import { ShareButtonsService } from './services/share-buttons.service';
import { WindowService } from './services/window.service';
import { NFormatterPipe } from './helpers/n-formatter.pipe';
import { ShareButton, ShareArgs, ShareProvider } from './helpers';
export var ShareButtonsModule = (function () {
    function ShareButtonsModule() {
    }
    ShareButtonsModule.forRoot = function () {
        return {
            ngModule: ShareButtonsModule,
            providers: [
                ShareButtonsService,
                WindowService
            ]
        };
    };
    ShareButtonsModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [
                        ShareButtonsComponent,
                        ShareButtonComponent,
                        ShareButtonDirective,
                        NFormatterPipe
                    ],
                    imports: [
                        CommonModule,
                        JsonpModule
                    ],
                    exports: [
                        ShareButtonsComponent,
                        ShareButtonComponent,
                        ShareButtonDirective,
                        NFormatterPipe
                    ]
                },] },
    ];
    /** @nocollapse */
    ShareButtonsModule.ctorParameters = [];
    return ShareButtonsModule;
}());
export { ShareButtonsComponent, ShareButtonDirective, ShareButton, NFormatterPipe, ShareButtonsService, ShareArgs, ShareProvider };
