// ng2-spin-kit/app/spinners.ts
import { RotatingPlaneComponent } from './spinner/rotating-plane.component'
import { DoubleBounceComponent } from './spinner/double-bounce.component'
import { WaveComponent } from './spinner/wave'
import { WanderingCubesComponent } from './spinner/wandering-cubes'
import { PulseComponent } from './spinner/pulse'
import { ChasingDotsComponent } from './spinner/chasing-dots'
import { CircleComponent } from './spinner/circle'
import { ThreeBounceComponent } from './spinner/three-bounce'
import { CubeGridComponent } from './spinner/cube-grid'
import { WordPressComponent } from './spinner/word-press'
import { FadingCircleComponent } from './spinner/fading-circle'
import { FoldingCubeComponent } from './spinner/folding-cube'

export {
  RotatingPlaneComponent,
  DoubleBounceComponent,
  WaveComponent,
  WanderingCubesComponent,
  PulseComponent,
  ChasingDotsComponent,
  CircleComponent,
  ThreeBounceComponent,
  CubeGridComponent,
  WordPressComponent,
  FadingCircleComponent,
  FoldingCubeComponent
};
