// ng2-sharebuttons/components/share-button/share-button.component.js
import { Component, Input, Output, EventEmitter, ChangeDetectionStrategy } from '@angular/core';
export var ShareButtonComponent = (function () {
    function ShareButtonComponent() {
        /** Show count, disabled by default */
        this.showCount = false;
        /** Output button count to calculate total share counts */
        this.count = new EventEmitter();
        /** Output pop up closed*/
        this.popUpClosed = new EventEmitter();
    }
    ShareButtonComponent.prototype.counter = function (count) {
        this.shareCount = count;
        this.count.emit(count);
    };
    /** emits closed button type: so user can tell which button has been clicked */
    ShareButtonComponent.prototype.shareClosed = function (provider) {
        this.popUpClosed.emit(provider);
    };
    ShareButtonComponent.decorators = [
        { type: Component, args: [{
                    selector: 'share-button',
                    template: "\n      <button [shareButton]=\"button.provider\"\n              [sbUrl]=\"url\"\n              [sbImage]=\"image\"\n              [sbTitle]=\"title\"\n              [sbDescription]=\"description\"\n              [sbTags]=\"tags\"\n              [sbShowCount]=\"showCount\"\n              (sbCount)=\"counter($event)\"\n              (sbPopUpClosed)=\"shareClosed($event)\"\n              [class.sb-show-count]=\"showCount\"\n        >\n\n        <div class=\"sb-template\" [innerHtml]=\"button.template\"></div>\n        <span *ngIf=\"showCount && shareCount\" class=\"sb-count\">{{ shareCount | nFormatter: 1 }}</span>\n      </button>\n    ",
                    changeDetection: ChangeDetectionStrategy.OnPush
                },] },
    ];
    /** @nocollapse */
    ShareButtonComponent.ctorParameters = [];
    ShareButtonComponent.propDecorators = {
        'url': [{ type: Input },],
        'title': [{ type: Input },],
        'description': [{ type: Input },],
        'image': [{ type: Input },],
        'tags': [{ type: Input },],
        'button': [{ type: Input },],
        'showCount': [{ type: Input },],
        'count': [{ type: Output },],
        'popUpClosed': [{ type: Output },],
    };
    return ShareButtonComponent;
}());
